framework/src/media/audio/audio_manager.c

```diff
@@ -682,6 +681,33 @@ static audio_manager_result_t set_audio_volume(audio_io_direction_t direct, uint
 
 audio_manager_result_t audio_manager_init(void)
 {
+	static int am_initialized = 0;
+	int card_id;
+
+	if (am_initialized) {
+		return AUDIO_MANAGER_SUCCESS;
+	}
+
+	am_initialized = 1;
+
+	g_audio_in_cards = (audio_card_info_t *)calloc(CONFIG_AUDIO_MAX_INPUT_CARD_NUM, sizeof(audio_card_info_t));
+	if (g_audio_in_cards == NULL) {
+		return AUDIO_MANAGER_OPERATION_FAIL;
+	}
+
+	for (card_id = 0; card_id < CONFIG_AUDIO_MAX_INPUT_CARD_NUM; card_id++) {
+		pthread_mutex_init(&(g_audio_in_cards[card_id].card_mutex), NULL);
+	}
+
+	g_audio_out_cards = (audio_card_info_t *)calloc(CONFIG_AUDIO_MAX_OUTPUT_CARD_NUM, sizeof(audio_card_info_t));
+	if (g_audio_out_cards == NULL) {
+		return AUDIO_MANAGER_OPERATION_FAIL;
```

Comment:
Is it okay not to free the g_audio_in_cards?

Reply:
I'll fix it